### ISSUE TITLE
Error in compiling images in less 

### ISSUE THREAD

**Issue by ntern** [vuejs/vue-loader#362]
`ERROR in ./src/assets/less/images/background.jpg
Module build failed: Error: parseQuery should get a string as first argument
    at Object.parseQuery (/var/www/icarespa/node_modules/loader-utils/index.js:68:9)
    at Object.module.exports (/var/www/icarespa/node_modules/url-loader/index.js:9:26)
 @ ./~/css-loader?sourceMap!./~/vue-loader/lib/style-rewriter.js?id=data-v-1!./~/less-loader!./src/assets/less/theme.less 6:244932-244966
 @ ./~/vue-style-loader!./~/css-loader?sourceMap!./~/vue-loader/lib/style-rewriter.js?id=data-v-1!./~/less-loader!./src/assets/less/theme.less
 @ ./src/App.vue
 @ ./src/app.js
 @ ./src/main.js
 @ multi app
`

**Comment by LinusBorg** [comment]
This is likely not a problem wqith vue-loder, as vue-loader itself does not handle images/assets.

But we cant tell without code & webpack config.

**Comment by ntern** [comment]
@LinusBorg So it could be the url-loader or something.
my webpack is setup like:

`const path = require('path')
const webpack = require('webpack')
var config = require('./config');
var utils = require('./utils');
const vueConfig = require('./vue-loader.config')

module.exports = {
    devtool: '#source-map',
    entry: {
        app: './src/app.js',
        vendor: ['vue', 'vue-router', 'vuex', 'vue-resource', 'vuex-router-sync', 'nprogress']
    },
    output: {
        path: config.build.assetsRoot,
        publicPath: process.env.NODE_ENV === 'production' ? config.build.assetsPublicPath : config.dev.assetsPublicPath,
        filename: '[name].js'
    },
    resolve: {
        extensions: ['.js', '.vue'],
        alias: {
            'src': path.resolve(__dirname, '../src'),
            'assets': path.resolve(__dirname, '../src/assets'),
            'components': path.resolve(__dirname, '../src/components')
        }
    },
    module: {
        loaders: [
            {
                test: /\.vue$/,
                loader: 'vue'
            },
            {
                test: /\.js$/,
                loader: 'babel',
                exclude: /node_modules/
            },
            {
                test: /\.json$/,
                loader: 'json'
            },
            {
                test: /\.html$/,
                loader: 'vue-html'
            },
            {
                test: /\.(png|jpe?g|gif|svg)(\?.*)?$/,
                loader: 'url',
                query: {
                    limit: 10000,
                    name: utils.assetsPath('images/[name].[hash:7].[ext]')
                }
            },
            {
                test: /\.(woff2?|eot|ttf|otf)(\?.*)?$/,
                loader: 'url',
                query: {
                    limit: 10000,
                    name: utils.assetsPath('fonts/[name].[hash:7].[ext]')
                }
            }
        ]
    },
    plugins: [
        new webpack.LoaderOptionsPlugin({
            vue: vueConfig
        })
    ]
}`

**Comment by LinusBorg** [comment]
are you sure the relative paths are correct? what is the actual less code?

**Comment by emahuni** [comment]
I have come across this today though I am using sass. It's vue-loader that's not working properly or changed some convention, which makes it a breaking change. I don't know what changed in vue-loader, but I was able to fix it in a few minutes. 

Here is my scenerio:

Before i installed vue-loader tagged `next` in npm everything was working perfectly. I was using another vue-loader version that works with VueJS 1.x. and decided to try Vue 2.0.0-rc.8 and ran into some problems about runtime and standalone builds blah blah. I fixed that as detailed [here](https://github.com/vuejs/vue/wiki/Vue-2.0-RC-Starter-Resources) by  installing vue-loader@next and putting the webpack cofiguration as required so that my .vue files that have <template> code can be properly compiled by webpack.

I then noticed that all images, luckily expect 1, were not loading correctly as if they were not being loaded correctly. I then looked at the code of images that were in the same file with the one that was loading correctly, it was as follows:

```html
<template>
<div class="front">
					<div class="cover">
						<img src="assets/images/thumb3.png"/>
					</div>
					<div class="user">
						<img class="img-circle" src="../../images/user.png"/>
					</div>
					<div class="content">
						<div class="main">
							<h4 class="name">Emmanuel Mahuni</h4>
							<p class="profession">Software Engineer</p>
						</div>
					</div>
				</div>
</template>
```

The second image is the one that was working properly. The other is an example of many...

It seemed like the path `assets/images/thumb3.png` is not being detected as a relative path, but `../../images/user.png` is.

So I just went right ahead and prefixed the other notorious path with a `./` to make it another type of relative path like `require` of common js requires. eg: 


```html
<template>
<div class="front">
					<div class="cover">
						<img src="./assets/images/thumb3.png"/>
					</div>
					<div class="user">
						<img class="img-circle" src="../../images/user.png"/>
					</div>
					<div class="content">
						<div class="main">
							<h4 class="name">Emmanuel Mahuni</h4>
							<p class="profession">Software Engineer</p>
						</div>
					</div>
				</div>
</template>
```

... and boom it worked. Notice, I never touched image-webpack-loader, which is responsible for loading image assets, just vue-loader and vue. Most of the other changes were `Vue = require ('vue');` to `import Vue from 'vue'; window.Vue = Vue;` etc so that my scripts are done through import instead of require, which i think may have something to do with it too... but generally that was the solution to the problem.
